### ISSUE TITLE
aiohttp >= 0.21 doesn't support match() on routes

### ISSUE THREAD

**Issue by rutsky** [aio-libs/aiohttp#766]
Router refactoring documentation [says](https://github.com/KeepSafe/aiohttp/blame/master/docs/new_router.rst#L73):

``` rst
``resource.match(...)`` is supported as well (while we believe it's not used widely).
```

However `match()` doesn't implemented on new Routes (and on Resources too):

```
from aiohttp import web

app = web.Application()

route = app.router.add_route("GET", "/hello", lambda _: None)
# Fails on aiohttp 0.21.0, works on 0.20.2.
route.match("/test")

resource = route.resource
# Fails on aiohttp 0.21.0, however resource.match() is not required for backward compatibility
resource.match("/test")
```

I think this backward incompatibility of 0.21 is not required and deprecated `match()` methods should be introduced back (for Routes, there is no need for them for Resources).

**Comment by rutsky** [comment]
This is one of the reasons of aiohttp_cors not working in 0.21: https://github.com/aio-libs/aiohttp_cors/issues/30

**Comment by rutsky** [comment]
After discussion on the mailing list it was decided that `.match()` should not be resurrected.
Fix of it's removal in docs done in #768.
